src/codeu/chat/util/encryption/Encryptor.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Encryptor.java
+ * Author: Arunabh Singh
+ * Main code of encrypting the actual String sequences produced by compression code.
+ */
+
+import java.security.SecureRandom;
+import java.math.BigInteger;
+
+public class Encryptor {
+    final int bitLength = 1024;
+    private BigInteger privateKey;
+    public PublicKey publicKey;
+
+    public Encryptor() {
+        newKeys();
+    }
+
+    public BigInteger encrypt(PublicKey key, String input) {
```

Comment:
It looks like you're using asymmetric encryption. Is this for encryption at rest (i.e. storage) or for encryption between 2 parties (i.e. client and server)?